cmd/kubepkg/templates/latest/deb/kubelet/lib/systemd/system/kubelet.service

```diff
@@ -1,8 +1,9 @@
 [Unit]
 Description=kubelet: The Kubernetes Node Agent
 Documentation=https://kubernetes.io/docs/home/
-Wants=network-online.target
-After=network-online.target
+Wants=network-online.target time-sync.target
+After=network-online.target time-sync.target
+RequiresMountsFor=/var/lib/kubelet
```

Comment:
i've seen some users that do that, but *not everyone* is mounting /var/lib/kubelet.
from my understanding, if the path is missing on initial Node setup (which is quite plausible) and if it never ends as a mount point, RequiresMountsFor would block the service start completely until something creates the path.

kubeadm pre-creates this path, so the service will start properly when kubeadm is used, but this is a breaking change for users out there that do not rely on the kubelet package without the kubeadm package.

also this pins the kubelet root directory on the side of the unit file, while the directory is configurable by the user via the "--root-dir" flag of the kubelet binary.

Reply:
In case a power user install kubelet by .deb/.rpm but target to use non-standard ‘--root-dir’, shall we assume that they also understand what are they doing and so the ‘kubelet.service’ need to update accordingly?

BTW don’t we create ‘/var/lib/kubelet’ during kubelet .deb/.rpm installation? I really don’t expect that (well I always install kubectl/kubelet/kubeadm with static binary and Ansible for cross OS compatibility, where always create ‘/var/lib/kubelet’ inside kubelet Ansible Role, see https://github.com/alvistack/ansible-role-kubelet/blob/master/tasks/main.yml#L64)...

Reply:
> In case a power user install kubelet by .deb/.rpm but target to use non-standard ‘--root-dir’, shall we assume that they also understand what are they doing and so the ‘kubelet.service’ need to update accordingly?

possibly, but this change could break their setup.
alternatively, other power have the option to add the `RequiresMountsFor=/var/lib/kubelet` condition optionally.

> BTW don’t we create ‘/var/lib/kubelet’ during kubelet .deb/.rpm installation?

i don't think so.

if kubeadm does not create the directory, it's the kubelet that creates it., but if the service starts the kubelet and the service is blocking until the directory exists the kubelet will never start.

Reply:
From https://github.com/kubernetes/release/blob/master/packages/rpm/kubelet.spec#L116-L121:
```
install -m 755 -d %{buildroot}%{_unitdir}
install -m 755 -d %{buildroot}%{_unitdir}/kubelet.service.d/
install -m 755 -d %{buildroot}%{_bindir}
install -m 755 -d %{buildroot}%{_sysconfdir}/cni/net.d/
install -m 755 -d %{buildroot}%{_sysconfdir}/kubernetes/manifests/
install -m 755 -d %{buildroot}/var/lib/kubelet/
```

From this point of view .deb should also sync with similar behavior, and so now `packages/deb/bionic/kubelet/debian/postinst` updated as:
```
mkdir -p /etc/cni/net.d
mkdir -p /etc/kubernetes/manifests
mkdir -p /etc/kubernetes/manifests
mkdir -p /lib/systemd/system/kubelet.service.d
mkdir -p /var/lib/kubelet
```

Shouldn't this be a better default and suitable for upgrade, too?

Reply:
i really like us to stop creating some of these folders in specs and leave it to the runtime, given the components can pick these locations with flags.

for example, in the future kubeadm has some plans to make `/etc/kubernetes/manifests` configurable...

Reply:
@neolit123 I give some checking to other packaging example, e.g.:
- https://salsa.debian.org/go-team/packages/etcd/-/blob/master/debian/etcd-server.postinst#L7-10
- https://salsa.debian.org/postgresql/postgresql-common/-/blob/master/debian/postgresql-common.dirs#L3
- https://salsa.debian.org/mariadb-team/mysql/-/blob/mysql-8.0/debian/master/debian/mysql-server-8.0.postinst#L195-200

Both of them are securing a reasonable default data directory for its own daemon installation with .deb, so a simple `apt -y install etcd` could always result with a running daemon without additional documentation nor configuration DURING initial installation.

Shall we keep in simple and stupid for most normal user, where allowing power user to configure it afterward as they wished?

Reply:
`RequiresMountsFor=/var/lib/kubelet` removed